synapse/lib/cell.py

```diff
@@ -1202,75 +1201,69 @@ async def _execBackupTask(self, dirn):
         mypipe, child_pipe = ctx.Pipe()
         paths = [str(slab.path) for slab in slabs]
         loglevel = logger.getEffectiveLevel()
+        proc = None
 
-        def spawnproc():
-            logger.debug('Starting multiprocessing target')
-            proc = ctx.Process(target=self._backupProc, args=(child_pipe, self.dirn, dirn, paths, loglevel))
-            proc.start()
-            hasdata = mypipe.poll(timeout=self.BACKUP_SPAWN_TIMEOUT)
-            if not hasdata:
-                raise s_exc.SynErr(mesg='backup subprocess stuck starting')
-            data = mypipe.recv()
-            assert data == 'ready'
-            return proc
+        try:
 
-        proc = await s_coro.executor(spawnproc)
+            async with self.nexsroot.applylock:
 
-        logger.debug('Syncing LMDB Slabs')
-        while True:
-            await s_lmdbslab.Slab.syncLoopOnce()
-            if not any(slab.dirty for slab in slabs):
-                break
+                logger.debug('Syncing LMDB Slabs')
 
-        try:
+                while True:
+                    await s_lmdbslab.Slab.syncLoopOnce()
+                    if not any(slab.dirty for slab in slabs):
+                        break
 
-            logger.debug('Acquiring LMDB txns')
-            mypipe.send('proceed')
+                logger.debug('Starting backup process')
 
-            # This is technically pending the ioloop waiting for the backup process to acquire a bunch of
-            # transactions.  We're effectively locking out new write requests the brute force way.
-            hasdata = mypipe.poll(timeout=self.BACKUP_ACQUIRE_TIMEOUT)
-            if not hasdata:
-                raise s_exc.SynErr(mesg='backup subprocess stuck acquiring LMDB txns')
+                args = (child_pipe, self.dirn, dirn, paths, loglevel)
 
-            data = mypipe.recv()
-            assert data == 'captured'
+                def waitforproc1():
+                    nonlocal proc
```

Comment:
Can we return the proc object ?

Reply:
This seems like a good simplification. :)

Reply:
That won't kill the process if it gets stuck in waitforproc1.